### PR TITLE
For #26289 - Remove tab groups metrics and setting

### LINKED ISSUE #26227: [Bug]: "Jump back in" section displays tab groups but tabs tray no longer supports them
### Steps to reproduce

1. Trigger a search
2. Open a few search results in tabs
3. Go to home screen

### Expected behaviour

Jump back in should display last tab, but no tab group

### Actual behaviour

Jump back in shows tab group. Tapping on it opens the tabs tray which no longer supports groups and therefore doesn't highlight the group. It also may, depending on the previously selected tab, highlight an unrelated tab.

### Device name

Pixel 6a

### Android version

Android 12

### Firefox release type

Firefox Nightly

### Firefox version

105.0a1

### Device logs

_No response_

### Additional information

Tab groups were removed as part of https://github.com/mozilla-mobile/fenix/issues/21898 in https://github.com/mozilla-mobile/fenix/commit/7e59a644d5890c1ce083b9435e27acd58d04fb1f. However, the "Jump back in" section wasn't addressed as part of this work, leaving this in an inconsistent state.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21139)

### LINKED ISSUE #26289: Remove search term tab groups setting and metrics
Follow-up from https://github.com/mozilla-mobile/fenix/pull/26003#issuecomment-1185724932

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21182)

### PR DESCRIPTION
**Pull request by MozillaNoah**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26289

### CODE REVIEW

**Comment by mavduevskiy** [comment]
@MozillaNoah 
[Failed test](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/6878181598860743499/details?stepId=bs.b75bb83da63ff7d4&testCaseId=1) looks legit, it looks like we have hidden the Search Group element from the settings, which the test is trying to find
Screen Shot 2022-08-02 at 1 25 43 PM

File: app/metrics.yaml [hunk 1]
```diff
@@ -7328,26 +7315,6 @@ recent_bookmarks:
     expires: 107
     unit: integer
 
-recent_searches:
```
Comment (csadilek) [inline app/metrics.yaml:7331]: The recently visited section is based on history, not tab groups, so this needs to stay if we want to retain groups in history.

File: app/metrics.yaml [hunk 2]
```diff
@@ -8015,19 +7903,6 @@ recently_visited_homepage:
     data_sensitivity:
       - interaction
     expires: 114
-  search_group_opened:
```
Comment (csadilek) [inline app/metrics.yaml:8018]: If this is also used when opening a group from "recently visisted" which navigates to history, then we need to keep it.

**Review by csadilek — CHANGES_REQUESTED** [review]
Removing the setting which is specifically for tab groups under the tabs settings, and the related telemetry is fine, but recently visited is based on history and independent of this change. Let's not remove those parts.

File: app/metrics.yaml (hunk 1 above)
Comment (MozillaNoah) [reply app/metrics.yaml]: ✅

File: app/metrics.yaml (hunk 2 above)
Comment (MozillaNoah) [reply app/metrics.yaml]: ✅

**Review by rocketsroger — APPROVED** [review]
Looks good. Thanks

**Pull request merged**